tests/09-setuidgid.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/sh
+
+### Constants
+c_valgrind_min=1
+test_stderr="${s_basename}-stderr.txt"
+
+# Keep this in sync with the value in tests/setuidgid/main.c!
+EX_NOPERM=77
+
+test_setuidgid_username_groupname() {
+	pass_exitcode=$1
+	skip_ok=$2
+	username_groupname=$3
+
+	setup_check_variables
+
+	printf "=== check ${s_count}\n" >> ${test_stderr}
+	cd ${scriptdir}/setuidgid/ && \
+	    ${c_valgrind_cmd} ./test_setuidgid "${username_groupname}"	\
+	    2>> ${test_stderr}
+	cmd_exitcode=$?
+
+	if [ ${cmd_exitcode} -eq "$EX_NOPERM" ]; then
+		if [ ${skip_ok} -eq "1" ]; then
+			# Skip this check, since it relies on us having
+			# permission to change the uid and/or gid.
+			echo "-1" > ${c_exitfile}
+		else
+			# Fail
+			echo "1" > ${c_exitfile}
+		fi
+	elif [ ${cmd_exitcode} -eq ${pass_exitcode} ]; then
+		# We match the expect output.
+		echo "0" > ${c_exitfile}
+	else
+		# Fail
+		echo ${cmd_exitcode} >> ${test_stderr}
+		echo "1" > ${c_exitfile}
+	fi
+}
+
+### Actual command
+scenario_cmd() {
+	rm -f ${test_stderr}
+
+	# These should all fail.  (Hopefully nobody actually has
+	# "fake_username" or "fake_groupname" on their systems!)
+	test_setuidgid_username_groupname 1 0 "nobody:"
+	test_setuidgid_username_groupname 1 1 "fake_username"
+	test_setuidgid_username_groupname 1 1 ":fake_groupname"
+	test_setuidgid_username_groupname 1 1 "fake_username:fake_groupname"
+
+	# Try as self.
+	test_setuidgid_username_groupname 0 1 "$(id -u -n):$(id -g -n)"
```

Comment:
Shouldn't this be `0 0`?  IIRC you should always be able to setuid/setgid to yourself?